enterprise/server/remote_execution/dirtools/dirtools.go

```diff
@@ -289,6 +289,86 @@ func uploadFiles(uploader *cachetools.BatchCASUploader, fc interfaces.FileCache,
 	return nil
 }
 
+// handleSymlink adds the symlink to the directory and actionResult proto so that
+// they could be recreated on the Bazel client side if needed.
+func handleSymlink(dirHelper *DirHelper, rootDir string, cmd *repb.Command, actionResult *repb.ActionResult, directory *repb.Directory, fqfn string) error {
+	target, err := os.Readlink(fqfn)
+	if err != nil {
+		return err
+	}
+	symlink := &repb.OutputSymlink{
+		Path:   trimPathPrefix(fqfn, rootDir),
+		Target: target,
+	}
+	foundSymlink := false
```

Comment:
I'm generally not a fan of using defers in this way because it's not always obvious that non-trivial code is being executed at the end of the function if you don't read the code carefully enough. You could instead define an inline function and call that instead of setting foundSymlink, e.g.

addSymlinkToDir := func() { ... }

Reply:
Great suggestion! It looks much cleaner now. 👍